hcl2template/parser.go

```diff
@@ -47,6 +48,10 @@ var packerBlockSchema = &hcl.BodySchema{
 // the parsed HCL and then return a []packer.Build. Packer will use that list
 // of Builds to run everything in order.
 type Parser struct {
+	CorePackerVersion *version.Version
+
+	CorePackerVersionString string
```

Comment:
Here we should trust that `CorePackerVersion` is correct and won't error when its passed in, so if we decide to go with moving hcl2template over to the sdk then I'd remove the later field:
```suggestion
	CorePackerVersion *version.Version
```

Reply:
I do trust the semantic version, but if we move the hcl2template to the SDK, CorePackerVersionString is for hcl variable interpolation of the packer.version() variable (in packer_config.go below)  				

`"version": cty.StringVal(cfg.CorePackerVersionString),`

This string includes the formatting like `-dev` and a commit hash which I believe is not present in the version.Version.

I think it's likely that we will be keeping hcl2template in the core, in which case these changes are unnecessary... that said I'm kind of in favor of leaving it this way at least for now since it will work regardless of whether we decide to extract hcl2template or leave it in the core.

Reply:
> This string includes the formatting like -dev and a commit hash which I believe is not present in the version.Version.

Ah, yes, I didn't think of that ! Yup that makes total sense to keep it ! So that we can display/save with exactly what version this was run upon.

Reply:
I wonder if we will have suprises with this and nighlty builds down the line where the version is  actually `v1.5.6-dev` but `packer.required_version` ( SemVer ) has no way to see/do-anything-about that `-dev` tag. 

But I mean those are nightly builds so we can expect advanced Packer users ? 🙂  Also that `required_version` can just be removed then.

Maybe we should just log something about it like: "hey you are using the `required_version` field but this is a dev build or something"

Reply:
Hmm, interesting. go-version does support -beta tags, at least according to the README: https://github.com/hashicorp/go-version#version-sorting

It's possible it'll already do the right thing if we just add a -dev tag to the semver invocation. We should tinker on it. Mind if I do that in a different PR?

Reply:
Oh, really cool ! Sure ! 🙂